### ISSUE TITLE
mobile imgur causes graphics error failures

### ISSUE THREAD

**Issue by classilla** [classilla/tenfourfox#407]
Open any imgur image (e.g., http://i.imgur.com/yeiMP1X.jpg )

As you scroll down, the page turns white. (Reloading works.) Log contains, repeatedly,

```
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)[GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)|[1][GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1056, 61427, 2549[GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1056, 61427, 2549
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)|[1][GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1056, 61427, 2549|[2][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)[GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)|[1][GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1056, 61427, 2549|[2][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2549)|[3][GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1056, 61427, 2549[GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1056, 61427, 2549
```

The actual numbers vary depending on window size. Not a regression from 45.9.

**Comment by classilla** [comment]
The error is generated by `gfx/layers/RotatedBuffer.cpp` but the actual failure seems to be in `gfx/thebes/gfxPlatform.cpp:gfxPlatform::CreateOffscreenContentDrawTarget`.

**Comment by classilla** [comment]
Switching to cairo does not fix the problem, and cairo seems to be failing as a backend with a related error:

```
Crash Annotation GraphicsCriticalError: |[0][GFX1]: Unknown image content format 0|[756][GFX1]: Unknown image content format 0|[757][GFX1]: Unknown image content format 0|[758][GFX1]: Unknown image content format 0|[759][GFX1]: Unknown image content format 1|[760][GFX1]: Unknown image content format 0[GFX1]: Unknown image content format 0
```

**Comment by classilla** [comment]
I think the problem is in `gfx/2d/Factory.cpp:Factory::CheckSurfaceSize`. We need to see which one of these conditions is not being met.

**Comment by classilla** [comment]
The extent limit is being exceeded.

```
  // reject images with sides bigger than limit
  if (extentLimit && (sz.width > extentLimit || sz.height > extentLimit)) {
fprintf(stderr, "failure 2 %i\n", extentLimit);
    gfxDebug() << "Surface size too large (exceeds extent limit)!";
    return false;
  }
```

```
failure 2 32767
failure 2 8192
Crash Annotation GraphicsCriticalError: |[0][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2487)|[26][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2387)|[22][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2387)|[23][GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1376, 61427, 2387|[24][GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2387)|[25][GFX1-]: Failed 2 buffer db=0x0 dw=0x0 for -59987, 1376, 61427, 2387[GFX1-]: Failed to allocate a surface due to invalid size Size(61427,2387)
```

**Comment by classilla** [comment]
We're running up against a hard CG limit, apparently.

**Comment by classilla** [comment]
This also happens in ESR 45.9 on 10.6, so it is not a 10.4/5 or PowerPC-specific problem.

**Comment by classilla** [comment]
However, imgur on ESR 45.9 yields a desktop view which does not have the autoscroll which causes this to fail, so they're sniffing the user agent and giving us the mobile view, which doesn't work. If I use a TenFourFox UA with 10.6 ESR 45.9, the same issue appears.

**Comment by classilla** [comment]
Sent a request to imgur to change their UA strategy.

**Comment by classilla** [comment]
Imgur never acquiesced, and I can't find a workaround otherwise, so closing as fixed by #422.
